### ISSUE TITLE
"Connection to server is erroring. Shutting down server." after update to Mathematica 12.3

### ISSUE THREAD

**Issue by bshoshany** [kenkangxgwe/vscode-lsp-wl#11]
I updated Mathematica to the new 12.3 release. I installed the paclets as instructed:

```mathematica
PacletInstall["CodeParser"]
PacletInstall["CodeInspector"]
```
I then entered the correct path to `wolfram.exe` in the extension settings:

```none
C:\Program Files\Wolfram Research\Mathematica\12.3\wolfram.exe
```

However, when I start VS Code and open a Mathematica source file, I get the following in the Output pane:

```none
[Error - 9:45:47 AM] Connection to server is erroring. Shutting down server.
[INFO  2021-05-21T09:45:47] Server listening from port 6536...
```

Is the extension incompatible with Mathematica 12.3? If so, could it be updated? I find this extension extremely useful, but don't want to downgrade Mathematica just so I can use it...

Any help would be appreciated.

**Comment by kenkangxgwe** [comment]
Hi Barak, is your [lsp-wl](https://github.com/kenkangxgwe/lsp-wl/) repo on the latest commit of the `master` branch?
Can you try `git pull` or even checkout the latest `develop` branch. Thanks!

**Comment by bshoshany** [comment]
Thanks for your quick reply. I tried it with the `develop` branch now, but unfortunately I still get the same error message. Any other suggestions?

**Comment by kenkangxgwe** [comment]
Just tried Wolfram Engine 12.3, confirmed that it doesn't work for me either. I found that they rewrite the ``ZeroMQLink` `` package in 12.3, so this should be an upstream bug. I have sent feedback to WRI and let's wait for a few days for them to reply.

**Comment by kenkangxgwe** [comment]
Hi Barak, could you take a look if [this](https://github.com/kenkangxgwe/lsp-wl/wiki#working-with-mathematica-123) will help you connect to the server as a workaround. Thanks!

**Comment by bshoshany** [comment]
Thanks, it's working now!

However, on my system, I don't have `ZeroMQLink` 1.2.1 installed, only 1.2.0 and 1.1.22 (as found using `PacletFind["ZeroMQLink"]`). After disabling 1.2.0 with `PacletDisable["ZeroMQLink" -> "1.2.0"]`, the VS Code extension now runs without problem.

**Comment by kenkangxgwe** [comment]
Great, glad that it worked for you!

**Comment by bshoshany** [comment]
Update: I started getting the "Connection to server is erroring" message again. Turns out `ZeroMQLink` has been automatically upgraded to 1.2.2, which again broke the language server. Disabling it with `PacletDisable["ZeroMQLink" -> "1.2.2"]` solved the problem. But I'm just wondering, are you in contact with Wolfram regarding the TCP issues in newer versions of the paclet? Seems like they updated the paclet but still didn't resolve the issues.
